expertise/service/utils.py

```diff
@@ -311,37 +314,58 @@ def _camel_to_snake(camel_str):
         # Handle Group cases
         config.match_group = starting_config.get('match_group', None)
         config.alternate_match_group = starting_config.get('alternate_match_group', None)
+        config.inclusion_inv = None
+        config.exclusion_inv = None
 
         if api_request.entityA['type'] == 'Group':
             config.match_group = [api_request.entityA['memberOf']]
+            edge_inv = api_request.entityA.get('expertise', None)
+
+            if edge_inv:
+                edge_inv_id = edge_inv.get('invitation', None)
+                if edge_inv_id is None or len(edge_inv_id) <= 0:
+                    raise openreview.OpenReviewException('Bad request: Expertise invitation indicated but ID not provided')
+                label = openreview_client.get_invitation(edge_inv_id).reply['content']['label']['value-radio'][0]
+                print(f"FETCHED LABEL")
+                if 'exclude' not in label.lower():
+                    config.inclusion_inv = edge_inv_id
+                else:
+                    config.exclusion_inv = edge_inv_id
+
         if api_request.entityB['type'] == 'Group':
             config.alternate_match_group = [api_request.entityB['memberOf']]
+            edge_inv = api_request.entityB.get('expertise', None)
+
+            if edge_inv:
+                edge_inv_id = edge_inv.get('invitation', None)
+                if edge_inv_id is None:
+                    raise openreview.OpenReviewException('Bad request: Expertise invitation indicated but ID not provided')
+                label = openreview_client.get_invitation(edge_inv_id).reply['content']['label']['value-radio'][0]
```

Comment:
when we give support to the new API, the invitation scheme will be different, like:

```
invitation.edge['content']['label']['param']['enum']
```

I also would try to use get() to avoid the KeyError exception. You could validate the format of the invitation and throw an error if it is not correct.